src/components/Header.tsx

```diff
@@ -0,0 +1,135 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import {
+  FC,
+  useState,
+  useRef,
+  useEffect,
+  SetStateAction,
+  Dispatch,
+  FormEvent,
+} from 'react';
+import classNames from 'classnames';
+
+import { CustomError } from '../types/CustomError';
+import { NewTodo, Todo } from '../types/Todo';
+
+import { postTodo, updateTodo } from '../api/todos';
+import { useLoadStatusContext } from '../utils/LoadStatusContext';
+
+import { initData } from '../constants/initData';
+
+type Props = {
+  todos: Todo[],
+  tempTodo: Todo | null,
+  setTodos: Dispatch<SetStateAction<Todo[]>>,
+  activeLeft: number,
+  setTempTodo: Dispatch<SetStateAction<Todo | null>>,
+  setError: (newError: CustomError, delay?: number) => void,
+};
+
+export const Header: FC<Props> = ({
+  todos,
+  tempTodo,
+  setTodos,
+  activeLeft,
+  setTempTodo,
+  setError,
+}) => {
+  const [newTodo, setNewTodo] = useState<NewTodo>(initData.newTodo);
+  const [inputDisabled, setInputDisabled] = useState(initData.inputDisabled);
+
+  const { setLoadingStatus } = useLoadStatusContext();
+  const focusInput = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (focusInput.current) {
+      focusInput.current.focus();
+    }
+  }, [tempTodo]);
+
+  const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    if (!newTodo.title.length) {
+      setError(CustomError.EmptyTitle, 3000);
+    } else {
+      setInputDisabled(true);
+      setTempTodo({ ...newTodo, id: 0 });
+
+      postTodo(newTodo)
+        .then((response) => {
+          setTodos(prevTodos => [...prevTodos, response]);
+          setTempTodo(null);
+          setNewTodo(initData.newTodo);
+          setInputDisabled(false);
+        })
+        .catch(() => setError(CustomError.Add, 3000));
+    }
+  };
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTodo((prevState) => ({ ...prevState, title: e.target.value }));
+  };
+
+  const handleToggleAll = () => {
+    let idsForToggle = todos.filter(todo => !todo.completed)
+      .map(todo => todo.id);
+
+    if (activeLeft === 0 || activeLeft === todos.length) {
+      idsForToggle = todos.map(todo => todo.id);
+    }
+
+    todos.forEach((todo, index) => {
+      if (idsForToggle.some(id => id === todo.id)) {
```

Comment:
Looks like you can iterate idsForToggle right away
```suggestion
    idsForToggle.forEach((todo, index) => {
```